openfga_sdk/api_client.py

```diff
@@ -178,8 +178,7 @@ async def __call_api(
         start = float(time.time())
 
         # header parameters
-        header_params = header_params or {}
-        header_params.update(self.default_headers)
+        header_params = {**self.default_headers, **(header_params or {})}
```

Comment:
Previously, default headers always won, and custom headers provided in request options were silently dropped.